graphql/src/main/java/io/micronaut/configuration/graphql/ws/GraphQLWsHandler.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.graphql.ws;
+
+import graphql.ExecutionResult;
+import io.micronaut.configuration.graphql.GraphQLConfiguration;
+import io.micronaut.configuration.graphql.GraphQLInvocation;
+import io.micronaut.configuration.graphql.GraphQLInvocationData;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.codec.CodecException;
+import io.micronaut.scheduling.ScheduledExecutorTaskScheduler;
+import io.micronaut.websocket.CloseReason;
+import io.micronaut.websocket.WebSocketSession;
+import io.micronaut.websocket.annotation.OnClose;
+import io.micronaut.websocket.annotation.OnError;
+import io.micronaut.websocket.annotation.OnMessage;
+import io.micronaut.websocket.annotation.OnOpen;
+import io.micronaut.websocket.annotation.ServerWebSocket;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+/**
+ * WebSocket request handler for the graphql-ws protocol. Supports the 'graphql-transport-ws' WebSocket subprotocol.
+ *
+ * @author Jeremy Grelle
+ * @since 4.0
+ */
+@ServerWebSocket(
+    value = "${" + GraphQLConfiguration.PREFIX + "." + GraphQLWsConfiguration.PATH_CONFIG + ":" + GraphQLWsConfiguration.DEFAULT_PATH + "}",
+    subprotocols = "graphql-transport-ws"
+)
+@Requires(property = GraphQLWsConfiguration.ENABLED_CONFIG, value = StringUtils.TRUE, defaultValue = StringUtils.FALSE)
```

Comment:
please, use a package-info.java with `@Configuration`. For example https://github.com/micronaut-projects/micronaut-security/blob/master/security-ldap/src/main/java/io/micronaut/security/ldap/package-info.java#L26